lib/msf/core/modules/metadata/obj.rb

```diff
@@ -203,11 +203,11 @@ def sort_platform_string
   end
 
   def force_encoding(encoding)
-    @name.force_encoding(encoding)
-    @fullname.force_encoding(encoding)
-    @description.force_encoding(encoding)
-    @author.each {|a| a.force_encoding(encoding)}
-    @references.each {|r| r.force_encoding(encoding)}
+    @name = @name.dup.force_encoding(encoding)
+    @fullname = @fullname.dup.force_encoding(encoding)
+    @description = @description.dup.force_encoding(encoding)
+    @author = @author.each {|a| a.dup.force_encoding(encoding)}
+    @references = @references.each {|r| r.dup.force_encoding(encoding)}
```

Comment:
Looks like this is mutating the original still, I think this is meant to be a map

```suggestion
    @author = @author.map {|a| a.dup.force_encoding(encoding)}
    @references = @references.map {|r| r.dup.force_encoding(encoding)}
```

Reply:
Ah, good point 👍 . Thanks @adfoster-r7